### PR TITLE
Ignore extra spaces in baseDN and attributes when creating LDAP search request

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@liggitt my hope is this fixes https://bugzilla.redhat.com/show_bug.cgi?id=1399781

**Comment by soltysh** [comment]
I don't know that much about LDAP, but seeing this:
```
uniqueMember: uid=user01, ou=users, dc=example, dc=com
uniqueMember: uid=user02,ou=users,dc=example, dc=com
uniqueMember: uid=user03, ou=users, dc=example, dc=com
```
seems not accurate. It's as if the server is having a bad day and putting spaces randomly in some entries and others not. I'm guessing it might be misconfiguration somewhere, but 100% sure.

File: pkg/auth/ldaputil/query.go [hunk 1]
```diff
@@ -104,11 +105,17 @@ type LDAPQueryOnAttribute struct {
...
 // the attribute to be filtered as well as any attributes that need to be recovered
 func (o *LDAPQueryOnAttribute) NewSearchRequest(attributeValue string, attributes []string) (*ldap.SearchRequest, error) {
+	removeSpaces := func(r rune) rune {
+		if unicode.IsSpace(r) {
+			return -1
+		}
+		return r
+	}
 	if strings.EqualFold(o.QueryAttribute, "dn") {
 		if _, err := ldap.ParseDN(attributeValue); err != nil {
```
Comment (soltysh) [inline pkg/auth/ldaputil/query.go]: Can we use result of this function instead of manually parsing and checking validity?

Comment (liggitt) [inline pkg/auth/ldaputil/query.go]: Agree, we should be comparing the structured forms

File: pkg/auth/ldaputil/query_test.go [hunk 2]
```diff
@@ -143,6 +143,33 @@ func TestNewSearchRequest(t *testing.T) {
...
+				SizeLimit:    DefaultSizeLimit,
+				TimeLimit:    DefaultTimeLimit,
+				TypesOnly:    DefaultTypesOnly,
+				Filter:       "(objectClass=*)",
+				Attributes:   append(DefaultAttributes, []string{"email", "phone"}...),
+				Controls:     DefaultControls,
+			},
+			expectedError: false,
+		},
+		{
```
Comment (liggitt) [inline pkg/auth/ldaputil/query_test.go]: Need a test that  ensures we still correctly identify mismatched white space inside the values as being out of scope

File: pkg/auth/ldaputil/query.go (hunk 1 above)
Comment (mfojtik) [inline pkg/auth/ldaputil/query.go]: +1

**Comment by mfojtik** [comment]
@enj fyi

**Comment by liggitt** [comment]
requires https://github.com/go-ldap/ldap/pull/98/

**Comment by liggitt** [comment]
closing in favor of https://github.com/openshift/origin/pull/12105

**Pull request closed without merging**